designs/modify-resources-attached-to-code-pipeline.md

```diff
@@ -0,0 +1,163 @@
+* the notification message has `Yes` and `No` buttons
+
+Proposed:
+* If the CloudFormation Stack is not associated with a CodePipeline, the same experience takes place.
+* If the CloudFormation Stack is associated with a CodePipeline, the user is given an alternate prompt (see [Confirmation Prompt](#confirmation-prompt) below). If the user agrees to proceed with Stack deletion, they do not get an additional prompt. The Stack is deleted.
+
+### Deploying a SAM Application to an existing CloudFormation Stack
+
+Users deploy a SAM Application using a Command Palette action. Through a Wizard, they are asked for details about what to deploy, and where to deploy it, including specifying a CloudFormation Stack. The stack specified can be new or may already exist. At this time, users must type the stack name in manually. There is a backlog task to replace this with a list that the user can choose a stack from.
+
+Proposed:
+After the wizard has collected all information from the user, we can check to see if the Stack belongs to a CodePipeline. If it does not, the deploy proceeds. If it does, the user is given a confirmation prompt (see [Confirmation Prompt](#confirmation-prompt) below). If the user agrees to proceed with the deploy, it proceeds normally. Otherwise, nothing happens - the wizard has already ended, and the user is back in the editor.
+
+Alternate:
+After the user enters a CloudFormation Stack, but before the wizard closes, we can check to see if the Stack belongs to a CodePipeline. If it does not, the wizard continues. If it does, the user is given a confirmation prompt, however this confirmation prompt is integrated with the wizard. If the user agrees to proceed with the deploy, the wizard moves on to the next step. Otherwise, the wizard moves back to the step where the user can specify a different CloudFormation Stack.
+
+### <a id="confirmation-prompt"></a>Confirmation Prompt
+
+The confirmation prompt contains the following messaging. See [Success criteria](#success-criteria) for the required information.
+
+---
+
+{ AWS Lambda Function | AWS CloudFormation Stack } '`arn_here`' is part of an AWS CodePipeline pipeline (`pipeline_arn_here`).
+
+Manual modification of resources outside of an AWS CodePipeline will skip any verification steps defined in the pipeline.
+
+Do you want to continue with this manual { deployment | deletion }?
+
+---
+
+The user will have the following actions/responses available:
+* `Yes` - proceeds with the manual operation
+* `No` - The default - Cancels the current operation (or goes back a step in the case of a wizard)
+* `Learn More` - This will link to user documentation (https://docs.aws.amazon.com/codepipeline/latest/userguide) but will not close the confirmation prompt
+
+Implementation
+==============
+
+Design
+------
+
+### Resources associated with a pipeline
+
+Resources that have a tag `aws:codepipeline:pipelineArn` are considered to have an association with a pipeline.
```

Comment:
Can we tag these resources with an alternate icon in the AWS Explorer? Maybe something like the existing icon with a pipe or a shield on it?

Reply:
I like this idea -- here are some additional thoughts to consider if we do provide this user experience:
* Is there a risk in the icon-state going stale?
* We'd have to query for this with each explorer refresh (separate from the get resources API calls). This may have performance impacts for users with many resources.

Reply:
How often do we update now? At the very least, we should update at that same cadence. We can also update on taking an action (since we'll be checking anyway), at the very least, we can update the single item.

I imagine that the risk of this getting stale is relatively low.

Reply:
+1. I like this idea. I'll add it in Jetbrains toolkit as well.